backend/src/validations/postType.validation.js

```diff
@@ -0,0 +1,26 @@
+const Joi = require('joi');
+const { objectId } = require('./custom.validation');
+
+const fieldName = Joi.string();
+
+exports.createPostType = {
+  body: Joi.object()
+    .keys({
+      communityId: Joi.string().custom(objectId).required(),
+      name: Joi.string().required(),
+      textFieldNames: Joi.array().items(fieldName).min(1).max(10).required(),
+      numberFieldNames: Joi.array().items(fieldName).min(1).max(10).required(),
+      dateFieldNames: Joi.array().items(fieldName).min(1).max(10).required(),
+      linkFieldNames: Joi.array().items(fieldName).min(1).max(10).required(),
+      locationFieldNames: Joi.array().items(fieldName).min(1).max(10).required(),
```

Comment:
Not every field type has to be on every post. So I think instead of` min(1)` it should be `min(0)`